judge/runguard.cc

```diff
@@ -836,6 +812,7 @@ void setrestrictions()
 	/* Set group-id (must be root for this, so before setting user). */
 	if ( use_group ) {
 		if ( setgid(rungid) ) error(errno,"cannot set group ID to `%d'",rungid);
+		gid_t aux_groups[10];
 		aux_groups[0] = rungid;
 		if ( setgroups(1, aux_groups) ) error(errno,"cannot clear auxiliary groups");
```

Comment:
Nitpick: there's no need to declare this as an array if only a single group is set, let alone one of length 10. Maybe for clarity an array is still better though:
```suggestion
		gid_t aux_groups[1];
		aux_groups[0] = rungid;
		if ( setgroups(1, aux_groups) ) error(errno,"cannot clear auxiliary groups");
```
Also, I guess we can just clear all auxiliary groups, since `rungid` is already set as the main group, so
```
		if ( setgroups(0, NULL) ) error(errno,"cannot clear auxiliary groups");
```
would suffice? (Since this is a C system call, I'm not sure the `NULL` can be replaced by `nullptr`.

Reply:
I think you are right, but suggest to move that change to its own PR.